Change GPL -> LGPL which better suits libraries

jjdmol opened the pull request:
This brings python-casacore in line with casacore. It also allows people to use python-casacore without being forced to make their own software GPL, which is not always possible.

tammojan commented:
Thanks for the PR! Could you change the GPL mention in setup.py as well?

tammojan commented:
For the most frequent contributors (notably @gervandiepen @gijzelaerr, @ludwigschwardt, @sjperkins), could you please respond to this PR with a thumbs-up?

gervandiepen commented:
Fine

tammojan commented:
@shibasisp Could you please also respond?

gijzelaerr commented:
sure! do it! libre software! MIT, BSD, LGPL, apache2, are all fine with me!

sjperkins commented:
I'm happy with this change.

ludwigschwardt commented:
Wow, I had assumed it was even more permissive like BSD. By all means go ahead! Definitely update `setup.py` too to keep PyPI happy.

aroffringa commented:
I would suggest to switch to LGPL version 3.0, and not to 2.1. The LGPL 3.0 is planned to be one of the possible default ASTRON licences. Version 2.1 is not compatible with Apache 2. I don't know in which situations that causes conflicts, but I don't see a reason to stay with 2.1 and have that possible compatibility issue.

Another suggestion is that the files should be inspected for license headers, and corrected if they don't match with the new license.

jjdmol commented:
Updated to LGPLv3, and updated setup.py. You may want to squash commits.

Note btw that the license mentioned by gitlab next to the code on https://github.com/casacore/python-casacore will auto update correctly once this hits master. In case anyone wondered.

ludwigschwardt commented:
Thanks! I wanted to remind you to update the classifiers in setup.py to reflect the new license but now see that there aren’t any 😂

shibasisp commented:
I'm happy with this change as well. :)

jjdmol commented:
The header & source files are now updated as well to refer to LGPL3.

tammojan commented:
Sorry to keep nagging... The headers mention the GNU Library General Public License, while the license is being changed to the GNU Lesser General Public License. They should have never made the abbreviation LGPL. Could you change Library to Lesser in the headers?

jjdmol commented:
Good catch. I've amended the commit.
